### PR TITLE
speedy: Compile new AnyException primitives

### PR DESCRIPTION
**Pull request by sofiafaro-da**
Part of #8020

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala [hunk 1]
```diff
@@ -1602,12 +1604,52 @@ private[lf] object SBuiltin {
...
           crash(s"AnyExceptionMessage applied to non-AnyException: $v")
       }
     }
   }
 
+  /** $any-exception-is-arithmetic-error :: AnyException -> Bool */
+  final case object SBAnyExceptionIsArithmeticError extends SBuiltinPure(1) {
+    override private[speedy] final def executePure(args: util.ArrayList[SValue]): SValue = {
+      args.get(0) match {
+        case SBuiltinException(ArithmeticError) =>
```
Comment (cocreature) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1625]: Have you considered adding a `getSException` similar to how we have `getSInt64`?

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala [hunk 2]
```diff
@@ -1602,12 +1604,52 @@ private[lf] object SBuiltin {
...
+  /** $any-exception-is-contract-error :: AnyException -> Bool */
+  final case object SBAnyExceptionIsContractError extends SBuiltinPure(1) {
+    override private[speedy] final def executePure(args: util.ArrayList[SValue]): SValue = {
+      args.get(0) match {
+        case SBuiltinException(ContractError) =>
+          SBool(True)
+        case SBuiltinException(_) =>
+          SBool(False)
+        case SAnyException(_, _) =>
+          SBool(False)
```
Comment (cocreature) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala]:
```suggestion
        case SBuiltinException(_) | SAnyException(_, _)=>
          SBool(False)
```

Comment (cocreature) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala]: same thing above

**Review by cocreature — APPROVED** [review]
Nice! Can we enable/add some tests that hit this?

**Comment by cocreature** [comment]
> Nice! Can we enable/add some tests that hit this?

Oh I guess we can’t because we don’t throw them yet.

**Comment by cocreature** [comment]
We could test them at the speedy level if we want.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala (hunk 2 above)
Comment (sofiafaro-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala]: Nice, thanks!

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala (hunk 1 above)
Comment (sofiafaro-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala:1625]: Nice idea, thanks :)

**Review by nickchapman-da — APPROVED** [review]
LGTM

**Comment by sofiafaro-da** [comment]
> We could test them at the speedy level if we want.

I'll wait to add tests since it'll just be easier to test them in Daml once we start throwing & catching these, and will probably be more stable then.

**Pull request merged**